qiskit/synthesis/linear/linear_matrix_utils.py

```diff
@@ -0,0 +1,146 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Utility functions for handling binary matrices."""
+
+import numpy as np
+from qiskit.exceptions import QiskitError
+
+
+def check_invertible_binary_matrix(mat: np.ndarray):
+    """Check that a binary matrix is invertible.
+
+    Args:
+        mat: a binary matrix.
+
+    Returns:
+        bool: True if mat in invertible and False otherwise.
+    """
+    if len(mat.shape) != 2 or mat.shape[0] != mat.shape[1]:
+        return False
+
+    mat = _gauss_elimination(mat)
+    rank = _compute_rank_after_gauss_elim(mat)
+    return rank == mat.shape[0]
+
+
+def random_invertible_binary_matrix(num_qubits: int, seed: np.random.Generator = None):
```

Comment:
Can `seed` be of type `int` here? If so, this should be probably reflecting in type hints.